lib/rubycritic/command_factory.rb

```diff
@@ -6,19 +6,28 @@ module RubyCritic
   class CommandFactory
     COMMAND_CLASS_MODES = %i[version help ci compare default].freeze
 
-    def self.create(options = {})
-      Config.set(options)
-      command_class(Config.mode).new(options)
-    end
+    class << self
+      def create(options = {})
+        Config.set(options)
+        command_class.new(options)
+      end
+
+      private
+
+      def command_class
+        require "rubycritic/commands/#{parsed_mode}"
+        Command.const_get(parsed_mode.capitalize)
+      end
+
+      def parsed_mode
+        return mode if COMMAND_CLASS_MODES.include? mode
+
+        :default
+      end
 
-    def self.command_class(mode)
-      mode = mode.to_s.split('_').first.to_sym
-      if COMMAND_CLASS_MODES.include? mode
-        require "rubycritic/commands/#{mode}"
-        Command.const_get(mode.capitalize)
-      else
-        require 'rubycritic/commands/default'
-        Command::Default
+      def mode
+        mode = Config.mode
+        mode.to_s.split('_').first.to_sym
```

Comment:
I wonder if we can add a new Config attribute called `command`, to use it like so `Config.command` that way we can delegate all this parse to a class and save it there.

Reply:
In the end, I went ahead and changed it, let's see what do you think about it.